backend/api/activity/index.js

```diff
@@ -6,68 +6,224 @@ const activity = express.Router();
+  .post(async (req, res) => {
+	//Ensure user is authenticated
     if (!req.user) {
-      return res.status(403).json({
+      return res.status(401).json({
         error: 'Not authenticated',
       });
     }
+	//Create the activity
     const statusCode = 200;
-    const currentDate = Date.now();
+    const currentDateTime = Date.now();
     const activitydata = new Activity({
-      title: req.body.title,
-      description: req.body.description,
-      creationDateTime: currentDate,
-      lastEditDateTime: currentDate,
+      title: req.body.title || "Unnamed activity",
+      description: req.body.description || "No description offered",
+      creationDateTime: currentDateTime,
+      lastEditDateTime: currentDateTime,
+	  startDateTime: req.body.startDateTime || (currentDateTime + 86400000),
+	  endDateTime: req.body.endDateTime || (currentDateTime + 90000000),
       postedBy: req.user.id,
     });
+	
+	//Ensure end time comes after start time
+	if (activitydata.endDateTime < activitydata.startDateTime) {
+	  return res.status(400).json({
+        error: 'endDateTime is less than startDateTime',
+      });
+	}
+	
     try {
+	  //Save it to the database
       await activitydata.save();
+	  
+	  //Response
       return res.status(statusCode).json({ activity: activitydata });
     } catch (error) {
+	  //Error Response
       return res.status(500).send(error);
     }
   });
 
 /**
  * Edits activity by id
  * TODO: Implement this
- * @param {Number} id - The id of the activity, must be only digits
+ * @param {Hex} id - The id of the activity, must be only hex digits
  */
-activity.route('/edit/:id([0-9]+)')
-  .post(passport.authenticate('local'), async (req, res) => {
-    if (!req.user) {
-      return res.status(403).json({
+activity.route('/edit/:id([a-f0-9]+)')
+  .post(async (req, res) => {
+	//Ensure user is authenticated
+	if (!req.user) {
+      return res.status(401).json({
         error: 'Not authenticated',
       });
     }
-    const statusCode = 200;
-    const shortened = Boolean(req.query.shortened);
-    return res.status(statusCode).json({
-      id: req.params.id,
-      status: statusCode,
-      shortened,
-    });
+	
+	try {
+	  //Get the activity to edit
+      const getobj = await Activity.findById(req.params.id).exec();
+	  
+	  //Ensure the authenticated user is author of this activity
+	  if (getobj.postedBy != req.user.id) {
+		return res.status(403).json({
+          error: 'Wrong account',
+        });
+	  }
+	  
+	  //Change the activity data
+      const statusCode = 200;
+      const shortened = Boolean(req.query.shortened);
+	  const currentDateTime = Date.now();
+	  getobj.title = req.body.title || getobj.title;
+	  getobj.description = req.body.description || getobj.description;
+	  getobj.startDateTime = req.body.startDateTime || getobj.startDateTime;
+	  getobj.endDateTime = req.body.endDateTime || getobj.endDateTime;
+	  getobj.lastEditDateTime = currentDateTime;
```

Comment:
When we're modifying a document from the database, if we don't modify a value it'll stay at what it was before so we don't need the short-circuit assignment for some of these.

Reply:
I'll retract my statement on this. Thinking about it further, this is probably the shortest way to go about assigning the new values.